File: typescript/agent/langgraph-fsi-agent/tools/creditScoreTool.ts
/**
 * A tool for retrieving information about credit scores from a financial services knowledge base.
 */
 import { tool } from "@langchain/core/tools";

/**
 * Define a tool for retrieving information about credit scores from a financial services knowledge base.
 */
// @ts-ignore - Complex type instantiation issue with LangChain tool function
export const creditScoreTool = tool(
    async (input: any): Promise<string> => {
        return "Your credit score is 550"; // Always return a credit score of 550 for testing purposes
    },
    {
        name: 'credit_score_tool',
        description: 'Returns a users credit score.',
    }
);

File: python/agent/langgraph-fsi-agent/before/src/galileo_langgraph_fsi_agent/tools/credit_score_tool.py
"""
A tool for retrieving information about credit scores from a financial services knowledge base.
"""

from typing_extensions import override

from langchain.tools import BaseTool


class CreditScoreTool(BaseTool):
    """
    CreditScoreTool is a tool for retrieving relevant information about credit scores from a financial services knowledge base.
    """

    name: str = "credit_score_retrieval"
    description: str = "Retrieve relevant information about credit scores from the financial services knowledge base"

    @override
    def _run(self) -> str:
        """
        Always return a credit score of 550 for testing purposes.
        """
        return "Your credit score is 550"
